array_api_tests/conftest.py

```diff
@@ -54,11 +54,6 @@ def pytest_addoption(parser):
         help="disable testing functions with output shapes dependent on input",
     )
     # CI
-    parser.addoption(
```

Comment:
Everyone using the test suite on CI is using this option. Should maybe leave this in for backwards compatibility.

Reply:
Yeah that makes sense, although a persisting concern of mine is the test suite is inherently pretty complex so it's worth being a bit ruthless in simplifying things wherever possible :thinking:

Reply:
We can deprecate it, although most people won't see the warning. I'd rather not break existing users if we don't have to.

Reply:
Deprecated instead, does feel right now!